Makefile

```diff
@@ -97,10 +97,11 @@ run_tests.lint: ## Runs the linter tests
 run_tests.backend: ## Runs the backend tests
 	$(MAKE) stop
 	docker compose up datastore dev-server redis firebase -d --no-deps
+	$(MAKE) update.requirements
 	@echo '------------------------------------------------------'
 	@echo '  Backend tests started....'
 	@echo '------------------------------------------------------'
-	$(SHELL_PREFIX) dev-server python -m scripts.run_backend_tests $(PYTHON_ARGS)
+	$(SHELL_PREFIX) dev-server sh -c "git config --global --add safe.directory /app/oppia && python -m scripts.run_backend_tests $(PYTHON_ARGS)"
```

Comment:
Why is `safe.directory` needed? Shouldn't `/app/oppia` have been created by the same user as is currently running?

Reply:
yes, it is the same user, but it was failing for the `check_backend_associated_test_file_test` IIRC. and after adding this fixes the issue. @gp201 can you please help why this issue is originating?

Reply:
@U8NWXD If I am correct then we need to create users within the Dockerfile, and we are not maintaining users. since the python script for the `check_backend_associated_test_file` script requires a git command.
we are similarly doing for lints checks -- https://github.com/oppia/oppia/blob/develop/Makefile#L121

Reply:
@Shivkant-Chauhan Could you please answer @U8NWXD's question directly?

Which user created `/app/oppia` and which user is currently running?

What error results if safe.directory is not use? What is the stack trace?